feat(icons): added `houses` icon

danielbayley opened the pull request:
<!-- Thank you for contributing! -->

<!-- Insert `closes #issueNumber` here if merging this PR will resolve an existing issue -->

## What is the purpose of this pull request?
- [x] New Icon

### Description
Added new `houses` icon.

### Icon use case
<!-- What is the purpose of this icon? For each icon added, please insert at least two real life use cases (the more the better). Text like "it's a car icon" is not accepted. -->
The plural of [`house`](https://lucide.dev/icons/house)…

### Alternative icon designs
<!-- If you have any alternative icon designs, please attach them here. -->

## Icon Design Checklist

### Concept
<!-- All of these requirements must be fulfilled. -->
- [x] I have provided valid use cases for each icon.
- [x] I have not added any a brand or logo icon.
- [x] I have not used any hate symbols.
- [x] I have not included any religious or political imagery.

### Author, credits & license
<!-- Please choose one of the following, and put an "x" next to it. -->
- [ ] The icons are solely my own creation.
- [x] The icons were originally created in #<issueNumber> by @danielbayley
- [x] I've based them on the following Lucide icons: `house`
- [ ] I've based them on the following design: <!-- provide source URL and license permitting use -->

### Naming
<!-- All of these requirements must be fulfilled. -->
- [x] I've read and followed the [naming conventions](https://lucide.dev/guide/design/icon-design-guide#naming-conventions)
- [x] I've named icons by what they are rather than their use case.
- [x] I've provided meta JSON files in `icons/[iconName].json`.

### Design
<!-- All of these requirements must be fulfilled. -->
- [ ] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [ ] I've made sure that the icons look sharp on low DPI displays.
- [ ] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [ ] I've made sure that the icons are visually centered.
- [ ] I've correctly optimized all icons to three points of precision.

## Before Submitting
- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [ ] I've checked if there was an existing PR that solves the same issue.

jguddas commented:
There should be a gap between the houses, and they should flow from bottom left to top right.

karsa-mistmere commented:
Hi @danielbayley, nice to have you back in our midst, and cute icon as usual. Can we work on the use cases though? I find that the list of tags is probably missing a few very trivial use cases (things such as village, community and neighborhood pop to mind, but I don't want to do all the work)

> There should be a 2px gap between the houses, and they should flow from bottom left to top right.

I don't think every single plural icon should follow the same template (and as previously stated I'm certainly not sold on your proposal of every single diagonal icon having to be bottom-left-to-top-right).

That being said, we could think about a name that's a better fit for this representation and does also leave room for a `houses` icon following your template. (Maybe `hice`? 🤪)

jguddas commented:
The bottom left to top right thing is honestly still a very WIP guideline, my idea is to be able to have a consistent spot in the bottom right for modifiers, so we can have houses-x or whatever.

danielbayley commented:
> Hi @danielbayley, nice to have you back in our midst

@karsa-mistmere Sure, how’re you guys doing?

> Can we work on the use cases though?

Ok, updated the above https://github.com/lucide-icons/lucide/pull/3241#issue-3070055849 with one obvious one, and also added some extra metadata…

> There should be a 2px gap between the houses

I think that would ruin the design in this case. That could work for a different version of the icon, with 2 houses in a different design, more like the `copy`[`-*`] icons…

> they should flow from bottom left to top right.

If you mean the path `d`irection, FYI @jguddas, Lucide [Studio](https://studio.lucide.dev) for some reason reveresed my `path` from Illustrator, which I think is a bug. I haven’t looked over the code for that, does it run on SVGO plugins? If so, I built a quick one to force a certain `path` `d`irection on processed SVGs, which I will publish soon. Would that be useful here?

danielbayley commented:
> > I think that would ruin the design in this case. That could work for a different version of the icon, with 2 houses in a different design, more like the copy[-*] icons…
> 
> Yeah, something like this maybe:
> 
> [<img alt="icons" width="200px" src="https://camo.githubusercontent.com/ac70c57720077c8adbf78ee1bd2a3dead38332cdb2b741998e2d2b369dd035bf/68747470733a2f2f6c75636964652e6465762f6170692f67682d69636f6e2f50484e325a776f67494868746247357a50534a6f644852774f693876643364334c6e637a4c6d39795a7938794d4441774c334e325a79494b4943423361575230614430694d6a516943694167614756705a326830505349794e43494b4943423261575633516d393450534977494441674d6a51674d6a5169436941675a6d6c7362443069626d39755a53494b4943427a64484a7661325539496d4e31636e4a6c626e5244623278766369494b4943427a64484a766132557464326c6b64476739496a496943694167633352796232746c4c577870626d566a59584139496e4a766457356b49676f6749484e30636d39725a5331736157356c616d3970626a3069636d3931626d5169436a344b49434138634746306143426b50534a744d5449754e6a6778494451754d6a51754f444d304c5334334d5456684d5334304e5341784c6a5131494441674d434178494445754f4467674d4777314c6a4135494451754d7a5930515445754e4455674d5334304e534177494441674d5341794d534135646a59754e545132595445754e4455674d5334304e534177494441674d533078494445754d7a67784969417650676f67494478775958526f49475139496b30784e5334304f4455674d5445754f446735515445754e4455674d5334304e534177494441674d5341784e6941784d3359324c6a55304e6b45784c6a51314e4341784c6a51314e434177494441674d5341784e4334314e4459674d6a46494e43347a4e6a52684d5334304e5451674d5334304e5451674d434177494445744d5334304e5451744d5334304e5452574d544e684d5334304e5341784c6a5131494441674d434178494334314d5455744d5334784d5446734e5334774f5330304c6a4d324e4745784c6a5131494445754e4455674d434177494445674d5334344f434177656949674c7a344b49434138634746306143426b50534a4e4e7934304d5341794d4334314e445a324c5452684d534178494441674d434178494445744d576779595445674d534177494441674d534178494446324e4349674c7a344b5043397a646d632b2e737667">Open lucide studio](https://studio.lucide.dev/edit?value=%3Cpath+d=%22m12.681+4.24.834-.715a1.45+1.45+0+0+1+1.88+0l5.09+4.364A1.45+1.45+0+0+1+21+9v6.546a1.45+1.45+0+0+1-1+1.381%22+/%3E%3Cpath+d=%22M15.485+11.889A1.45+1.45+0+0+1+16+13v6.546A1.454+1.454+0+0+1+14.546+21H4.364a1.454+1.454+0+0+1-1.454-1.454V13a1.45+1.45+0+0+1+.515-1.111l5.09-4.364a1.45+1.45+0+0+1+1.88+0z%22+/%3E%3Cpath+d=%22M7.41+20.546v-4a1+1+0+0+1+1-1h2a1+1+0+0+1+1+1v4%22+/%3E&name=house)

Yeah exactly (although it should probably be flipped horizontally, for consistency). I still prefer my version though, and which better depicts ‘many houses’. I’m cool with having both…

ericfennis commented on icons/houses.svg:
```diff
@@ -0,0 +1,14 @@
+<svg
+  xmlns="http://www.w3.org/2000/svg"
+  width="24"
+  height="24"
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M10 20v-5a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v5" />
+  <path d="M6 17H3a1 1 0 0 1-1-1V9.5c0-.6.3-1.2.7-1.6l4.6-3.7a1 1 0 0 1 1.3 0l8.7 7.3c.4.4.7.9.7 1.5v6a1 1 0 0 1-1 1H7a1 1 0 0 1-1-1v-6c0-.6.3-1.1.7-1.5l8.6-7.3a1 1 0 0 1 1.3 0l4.6 3.7c.5.4.8 1 .8 1.6V16a1 1 0 0 1-1 1h-3" />
+</svg>
```
Apply design from @jguddas 
```suggestion
<svg
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="m12.681 4.24.834-.715a1.45 1.45 0 011.88 0l5.09 4.364A1.45 1.45 0 0121 9v6.546a1.45 1.45 0 01-1 1.381" />
  <path d="M15.485 11.889A1.45 1.45 0 0116 13v6.546A1.454 1.454 0 0114.546 21H4.364a1.454 1.454 0 01-1.454-1.454V13a1.45 1.45 0 01.515-1.111l5.09-4.364a1.45 1.45 0 011.88 0z" />
  <path d="M7.41 20.546v-4a1 1 0 011-1h2a1 1 0 011 1v4" />
</svg>
```
